src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -644,9 +659,11 @@ module Make (Inputs : Inputs_intf.S) = struct
       | None ->
           Base.location_of_account (get_parent t) account_id
 
-    let location_of_account_batch =
-      self_find_or_batch_lookup self_find_location
-        Base.location_of_account_batch
+    let location_of_account_batch t =
+      (* TODO consider handling special case of empty addresses *)
```

Comment:
@mrmr1993 looks like this is needed. `self_find_or_batch_lookup` would end up accessing disk for empty accounts otherwise, no?

Reply:
@deepthiskumar I guess we don't have a choice in this function because unlike loading an account by location here we can't tell of an account whether it's a new one or just wasn't loaded from parent.

It might have an implication to performance of transaction processing (I do see some db lookups despite the fact that we preload a lot of stuff). But I think we'll handle it via special-casing `unsafe_preload` function, not right here.